ci/ray_ci/container.py

```diff
@@ -0,0 +1,84 @@
+import os
+import subprocess
+
+from typing import List
+
+from ci.ray_ci.utils import chunk_into_n
+
+DOCKER_ECR = "029272617770.dkr.ecr.us-west-2.amazonaws.com"
+DOCKER_REPO = "ci_base_images"
+DOCKER_TAG = f"oss-ci-build_{os.environ.get('BUILDKITE_COMMIT')}"
+
+
+def run_tests(
+    test_targets: List[str],
+    pre_run_commands: List[str],
+    parallelism,
+) -> int:
+    """
+    Run tests parallelly in docker
+    """
+    chunks = chunk_into_n(test_targets, parallelism)
+    runs = [_run_tests_in_docker(chunk, pre_run_commands) for chunk in chunks]
```

Comment:
how do we handle the test logs? will this mix all the logs together?

Reply:
they are printed in different chunks; still readable in term of what passed or failed, but can definitely be better.

Check https://buildkite.com/ray-project/oss-ci-build-pr/builds/30546#0189bdf1-cf6d-41ae-bcae-6ae71f73fc7d for example

Reply:
let me know if you want to improve this before merging

Reply:
yeah, it is now a bit hard to tell.. will probably become even more confusing when some test has errors to print out.

I do not think you have to handle it in this PR, but maybe leave a TODO or something.

either save the logs in files, and upload as artifacts, and you would need some way to print out the progress.

and/or still print them to stderr/stdout, but have a filter to prepend a stream id or something.

I also think there is no real guarantee that the log lines will be spit out in "chunks" or even "lines". I think that might be just an accidental result of bazel's ascii printing meeting the buildkite's log recording.

Reply:
you're right that there are no guarantee that logs are in chunk, it happens to work because these tests are slow ;)